Phoenix tracing local dev setup

ASCE-D opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Phoenix tracing enabled app-wide with automatic runtime initialization.
  * Support for both local and GitHub repo sources when accessing project code and diffs.
* **Bug Fixes**
  * Tracing initialization failures are non-fatal and emit warnings.
  * Change detection now returns consistent responses on failures and preserves patches.
* **Refactor**
  * Repo resolution and metadata extraction streamlined for clearer dev-mode behavior.
* **Documentation**
  * Added local Phoenix tracing setup guidance (view traces at http://localhost:6006; local dev only).
* **Chores**
  * Added pinned tracing/instrumentation packages and new env vars for tracing.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

ASCE-D committed:
tracing

ASCE-D committed:
update readme

ASCE-D committed:
Merge branch 'main' into feat-tracing

ASCE-D committed:
Merge b8f8adb47b97a9732b200411efe4a07c5816b7fc into 56adc4f76ef49999b3ee991e49322520f1c25e7d

github-actions[bot] committed:
chore: Auto-fix pre-commit issues

Dsantra92 commented on app/modules/intelligence/tracing/phoenix_tracer.py:
```diff
@@ -0,0 +1,215 @@
+logger = setup_logger(__name__)
+
+# Global flag to track if Phoenix is initialized
+_phoenix_initialized = False
```
Global variables should be in capital letters.
```suggestion
_PHOENIX_INITIALIZED = False
```

Dsantra92 commented on app/modules/intelligence/tracing/phoenix_tracer.py:
```diff
@@ -0,0 +1,215 @@
+            f"  Project: {project_name}\n"
+            f"  Endpoint: {endpoint}\n"
+            f"  Auto-instrument: {auto_instrument}"
+        )
```
Don't use f strings in logger.

Dsantra92 reviewed (commented):
Just some minor nitpicks about formatting. Everything looks good otherwise.

ASCE-D committed:
nitpicking

ASCE-D committed:
Merge fea51faa9109caf943975855cd434e5b1f595966 into 56adc4f76ef49999b3ee991e49322520f1c25e7d

github-actions[bot] committed:
chore: Auto-fix pre-commit issues

ASCE-D committed:
env instructions

ASCE-D committed:
Remove Phoenix API key from .env.template

Remove commented-out Phoenix API key from template

dhirenmathur committed:
Fix dev mode to support both GitHub and local repos (#477)

* Fix dev mode to support both GitHub and local repos

Previously in development mode, the system could not handle GitHub
repositories when GitHub credentials were configured. Two issues:

1. parsing_controller.py always converted repo_name to repo_path in dev mode
2. CodeProviderService always used LocalRepoService in dev mode

Changes:
- parsing_controller.py: Only clear repo_name if both repo_path and repo_name
  are provided (prioritize local). Otherwise keep whichever is provided.
- config_provider.py: Added is_github_configured() helper method
- code_provider_service.py: Changed from choosing one service globally to
  dynamically routing requests:
  * Stores both GithubService and LocalRepoService instances
  * get_repo(): Routes based on whether repo_name is a local path
  * get_project_structure_async(): Routes based on project's repo_path field
  * get_file_content(): Routes based on project's repo_path field

This allows development mode to work with both local repositories and
GitHub repositories when credentials are configured.

Fixes issue where dhirenmathur/test-mongo was treated as local path.

* chore: Auto-fix pre-commit issues

* Update cp client

* chore: Auto-fix pre-commit issues

---------

Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>

ASCE-D committed:
added a source variable

ASCE-D committed:
Merge a74a3692f13f0f8ab80d81b85f1f1e8b614cecbe into 185d0e543bc69a8d54d0413c671345a38fb9ffe9

github-actions[bot] committed:
chore: Auto-fix pre-commit issues
